### ISSUE TITLE
Pass add_requireconfs to packages when using xmake

### ISSUE THREAD

**Issue by SirLynix** [xmake-io/xmake#4726]
### Is your feature request related to a problem? Please describe.

when I write this:
```lua
add_requires("nazaraengine")
add_requireconfs("nazaraengine.**", { system = false })
```

it's not passed to the xmake used to build the package ([see the script here](https://github.com/NazaraEngine/xmake-repo/blob/master/packages/n/nazaraengine/xmake.lua)).

See the build log: https://github.com/DigitalPulseSoftware/ThisSpaceOfMine/actions/runs/7844246931/job/21406172638#step:10:1653

### Describe the solution you'd like

It would be nice that xmake passes requireconfs to xmake packages as well using XMAKE_RCFILES, for example:
```lua
add_requires("nazaraengine")
add_requireconfs("nazaraengine.**", { system = false })
```

would generate:
```
add_requireconfs("**", { system = false })
```
in a XMAKE_RCFILES

**Comment by waruqi** [comment]
pass_depconfs polify should will pass it. try set `override = true` or debug here

https://github.com/xmake-io/xmake/blob/cde78b29f0f7e0f45d237bb19d99b764133e2bac/xmake/modules/package/tools/xmake.lua#L444

**Comment by SirLynix** [comment]
> pass_depconfs policy should will pass it. try set `override = true` or debug here
> 
> https://github.com/xmake-io/xmake/blob/cde78b29f0f7e0f45d237bb19d99b764133e2bac/xmake/modules/package/tools/xmake.lua#L444

It does not because the packages are not listed in the package xmake script, those are private dependencies which the lib is statically linked to.

**Comment by waruqi** [comment]
but it has been merged into dep require info.

https://github.com/xmake-io/xmake/blob/cde78b29f0f7e0f45d237bb19d99b764133e2bac/xmake/modules/private/action/require/impl/package.lua#L630

**Comment by SirLynix** [comment]
It works only for packages listed as deps in the xmake build script, so I guess I need to write out all add_requires, even headeronly ones as `add_deps("..", { private = true })` in the package xmake.lua.

**Comment by waruqi** [comment]
I still don't understand what you mean, can you provide a simple example project that can be reproduced?

**Comment by SirLynix** [comment]
```lua
package("testpkg")
    on_install(function (package)
        io.writefile("xmake.lua", [[
add_requires("fmt", { configs = { headeronly = true } })

target("xxx")
    set_kind("$(kind)")
    add_packages("fmt")
    add_files("src.cpp")
]])
        import("package.tools.xmake").install(package)
    end)
package_end()

add_requires("testpkg")
add_requireconfs("testpkg.fmt", { system = false }) -- won't match because testpkg doesn't know it uses fmt

target("xx")
    add_packages("testpkg")
```

**Comment by waruqi** [comment]
thanks, I will look at it in these days.

**Comment by waruqi** [comment]
> Sure:
> 
> ```lua
> package("testpkg")
>     on_install(function (package)
>         io.writefile("xmake.lua", [[
> add_requires("fmt", { configs = { headeronly = true } })
> 
> target("xxx")
>     set_kind("$(kind)")
>     add_packages("fmt")
>     add_files("src.cpp")
> ]])
>         import("package.tools.xmake").install(package)
>     end)
> package_end()
> 
> add_requires("testpkg")
> add_requireconfs("testpkg.fmt", { system = false }) -- won't match because testpkg doesn't know it uses fmt
> 
> target("xx")
>     add_packages("testpkg")
> ```

you need add `add_deps("fmt")` in `package("testpkg")`

**Comment by SirLynix** [comment]
Yes that's what I said, a package needs to know about its private dependencies.

So I need to add a lot of `add_deps("xx", { private = true })` to my packages for this to work

**Comment by waruqi** [comment]
> Yes that's what I said, a package needs to know about its private dependencies.
> 
> So I need to add a lot of `add_deps("xx", { private = true })` to my packages for this to work

yes, you must add them.
